core/src/main/java/aima/core/logic/api/firstorderlogic/FOLKnowledgeBase.java

```diff
@@ -0,0 +1,62 @@
+package aima.core.logic.api.firstorderlogic;
+
+import aima.core.logic.basic.firstorder.kb.data.Literal;
+import aima.core.logic.basic.firstorder.parsing.ast.Term;
+import aima.core.logic.basic.firstorder.parsing.ast.Variable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An interface representing the First Order Logic (FOL) Knowledge Base
+ *
+ * @author samagra
+ */
+public interface FOLKnowledgeBase {
+    /**
+     * Adds the specified sentence to the knowledge base.
+     *
+     * @param sentence a fact to be added to the knowledge base.
+     */
+    void tell(String sentence);
+
+    /**
+     * Adds the specified sentences to the knowledge base.
+     *
+     * @param sentences A set of facts to be added to the knowledge base.
+     */
+    void tell(Set<? extends Literal> sentences);
+
+    /**
+     * @return A list of all the definite implication Clauses useful for inference
+     *      algorithms.
+     */
+    List<Clause> getAllDefiniteClauseImplications();
+
+
+    /**
+     * Used to check if a sentence is a <b>renaming</b> of a known fact. One sentence is a
```

Comment:
I'm not sure it is worth worrying about renaming (especially since the default implementation ignores the issue). COuldn't we do without this complication?